src/sw/redis++/reply.h

```diff
@@ -422,19 +422,22 @@ Variant<Args...> parse(ParseTag<Variant<Args...>>, redisReply &reply) {
 template <typename T, typename std::enable_if<IsSequenceContainer<T>::value, int>::type>
 T parse(ParseTag<T>, redisReply &reply) {
 #ifdef REDIS_PLUS_PLUS_RESP_VERSION_3
-    if (!is_array(reply) && !is_set(reply)) {
-        throw ParseError("ARRAY or SET", reply);
+    if (is_array(reply) || is_set(reply)) {
```

Comment:
This should be logically equivalent:

not P and not Q = not (P or Q). Reversing the if/then makes the valid branch: P or Q

As before, this is necessary for Nvidia to not throw a warning

Reply:
I think the root cause is that if 'REDIS_PLUS_PLUS_RESP_VERSION_3' is defined, there will be two `throw` clauses.

Can you revert changes in reply.h, and put `throw ParseError("ARRAY", reply);` before `#endif`? And check if the code pass the compiler?

```
 #ifdef REDIS_PLUS_PLUS_RESP_VERSION_3
     if (!is_array(reply) && !is_set(reply)) {
          throw ParseError("ARRAY or SET", reply);
 #else
     if (!is_array(reply)) {
-#endif
         throw ParseError("ARRAY", reply);
+#endif
     }
```